### PR TITLE
Add ability to configure opensearch snapshot repositories

### PR DESCRIPTION
**Pull request by dbason**
This PR does the following things:
- Creates a controller for managing Opensearch snapshot repositories
- Adds S3 configuration to the wrapper resource
- Adds s3 configuration to the Logging admin API

### CODE REVIEW

File: pkg/resources/opensearchrepository/opensearchrepository.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type Reconciler struct {
+	reconciler.ResourceReconciler
+	client      client.Client
+	respository *loggingv1beta1.OpensearchRepository
```
Comment (jaehnri) [inline pkg/resources/opensearchrepository/opensearchrepository.go:22]: there's a small typo here

Comment (jaehnri) [inline pkg/resources/opensearchrepository/opensearchrepository.go:22]: In the future, do we plan to support other types of repositories? Like using plain shared file system (shared file system as a snapshot repository, add it to opensearch.yml)

File: pkg/resources/opniopensearch/resources.go [hunk 2]
```diff
@@ -254,3 +282,49 @@ func calculateImage(repo, version, opensearchVersion string) string {
 		version,
 	)
 }
+
+func (r *Reconciler) pluginsList() []string {
+	plugins := []string{}
+	if r.instance.Spec.S3Settings != nil {
+		plugins = append(plugins, "repository-s3")
+	}
```
Comment (jaehnri) [inline pkg/resources/opniopensearch/resources.go:290]: Do we plan to also support the FileSystem snapshots in the future?

Comment (dbason) [reply pkg/resources/opniopensearch/resources.go:290]: It was pretty easy to implement so I have left the door open for that.  We may also support other storage providers besides S3

File: pkg/resources/opensearchrepository/opensearch.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+			OpensearchServiceName: cluster.Spec.General.ServiceName,
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	settings := osapi.RepositoryRequest{}
+	switch {
+	case r.respository.Spec.Settings.S3 != nil:
```
[the remark this answers is not in the document]
Comment (dbason) [reply pkg/resources/opensearchrepository/opensearch.go:36]: The logging plugin creates an OpniOpensearch CRD based on what is passed to its API.  The OpniOpensearch CRD then creates the additional resources required, including the OpensearchRepository.

File: pkg/resources/opniopensearch/resources.go (hunk 2 above)
Comment (jaehnri) [reply pkg/resources/opniopensearch/resources.go:290]: Cool, I'll pick it up whenever I can.

File: pkg/resources/opensearchrepository/opensearch.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (jaehnri) [reply pkg/resources/opensearchrepository/opensearch.go]: And who communicates to the LoggingAdmin API? Is it the Opni UI?

File: pkg/resources/opniopensearch/resources.go (hunk 2 above)
Comment (jaehnri) [reply pkg/resources/opniopensearch/resources.go:290]: I created https://github.com/rancher/opni/issues/1676 so it doesn't get lost.

File: pkg/resources/opensearchrepository/opensearch.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (dbason) [reply pkg/resources/opensearchrepository/opensearch.go]: Either the Admin UI or the CLI

**Review by kralicky — APPROVED** [review]

**Pull request merged**